File: go/examples/arrays.go
package main

import "fmt"

func main() {
	var a [5]int
	fmt.Println("empty:", a, "of len", len(a))

	// declare and init
	b := [5]int{1, 2, 3, 4, 5}
	fmt.Println(b)
}

File: kotlin/tour/src/andersc/tour/pro/arrays.kt
package andersc.tour.pro

fun main() {
    // empty array
    val empty = emptyArray<String>()
    println(empty)

    val strs = Array(size = 5, init = { index -> "Item #$index"})
    println(strs.toString())
    println(strs[0])

    val ints = intArrayOf(1, 2, 3)
    println(ints)

    // create using closure
    val a = Array(3) { i -> i * 2}

    // extensions
    println("average: ${a.average()}, first: ${a.firstOrNull()}")

    // iterate
    for (i in a) {
        print(i)
        print(' ')
    }
}